### PR TITLE
Bump rc 2.8.0 and pre-release v2.8.0rc1

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] ~Bumped the build number (if the version is unchanged)~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,38 +20,42 @@ build:
...
     - scipy >=1.6
     # Normally installed via pip:
     - cartopy
     - cftime
-    - cf-units >=3.0.0, <3.1.0, !=3.0.1.post0
+    - cf-units
     - dask
     - esgf-pyclient >=0.3.1
-    - esmpy !=8.1.0
+    - esmpy !=8.1.0,<8.4.0
```
Comment (valeriupredoi) [inline recipe/meta.yaml:39]: indeed this needs be kept until iris-esmf-regrid fix their stuff (I got at least one PR open there) - once this gets freed and we make prospector optional then Python=3.11 support comes as a bonus immediately :grin:

File: recipe/meta.yaml [hunk 2]
```diff
@@ -64,7 +68,8 @@ test:
     - setup.cfg
   requires:
     # - ESMValTool_sample_data==0.0.3
-    - flake8 <5.0
+    - flake8
+    - mypy <0.990
```
Comment (valeriupredoi) [inline recipe/meta.yaml:72]: we need the new pin `mypy>=0.990` here

File: recipe/meta.yaml [hunk 3]
```diff
@@ -81,7 +86,7 @@ test:
     # - types-PyYAML
   commands:
     - pip install ESMValTool-sample-data types-requests types-pkg_resources types-PyYAML
-    - pytest -n 2 --ignore=run_test.py --ignore=tests/unit/documentation/
+    - pytest -n 2 -m 'not mypy' --ignore=run_test.py --ignore=tests/unit/documentation/
```
Comment (valeriupredoi) [inline recipe/meta.yaml:89]: once we repin mypy >0.990 this can be removed

**Review by valeriupredoi — COMMENTED** [review]
looking sharp, man! Just the mypy wiggle then we good - can you pls make it RFR so I can approves and merge :beer:

**Comment by valeriupredoi** [comment]
great! But can we do it this way: this PR updates only the dependencies in the `rc` channel/branch and not the version to 2.8.0rc1, and sha, and in a separate PR we do the rc version update - it looks a bit messy history-wise and I am thinking we should keep things clean, especially if we pass this over to @remi-kazeroni afterwards, we'd like him to have less issues with this :grin:

**Comment by zklaus** [comment]
Hm. I am not sure I follow. The updating of the dependencies is only due to the updated version, right? To me, it doesn't make sense to separate this out.

**Comment by valeriupredoi** [comment]
Well, not quite, most of the commits are from updating to 2.7.0 - I reckon it'd be cleaner wrt GH history if we did first a 'refresh' then a version update. Let's talk at 12 today - @bouweandela you wanna join our quick Zoom call at 12 GMT, buds?

**Comment by valeriupredoi** [comment]
this is not looking right - and the solver is failing looking for a 3.11 ABI https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=669438&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=217 - is it not `, <3.11` the correct syntax?

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by valeriupredoi** [comment]
> Well, not quite, most of the commits are from updating to 2.7.0 - I reckon it'd be cleaner wrt GH history if we did first a 'refresh' then a version update. Let's talk at 12 today - @bouweandela you wanna join our quick Zoom call at 12 GMT, buds?

@zklaus @remi-kazeroni and myself, we talked about this and decided to go this way so we don't rebuild an older 2.7.0rcX with the new changes from `main` :+1:

**Comment by valeriupredoi** [comment]
@conda-forge-admin, please restart ci

**Pull request merged**